### PR TITLE
Fix bug when completing "ON parent." clauses

### LINKED ISSUE #40: Error when typing a query
I started writing this query:

```
SELECT d.* FROM doodads d JOIN trinkets t on d.
```

## `JOIN` doesn't appear

`JOIN` did not auto-complete as an option (I only saw `JSON`)

## Unhandled Exception

When I type the `.` at the end (I'm trying to join on `d.trinket_id = doodad.id`) I get the following output:

```
Unhandled exception in event loop:
  File "/home/admin/.local/lib/python3.7/site-packages/prompt_toolkit/eventloop/coroutine.py", line 92, in step_next
    new_f = coroutine.throw(exc)
  File "/home/admin/.local/lib/python3.7/site-packages/prompt_toolkit/buffer.py", line 1654, in new_coroutine
    yield From(coroutine(*a, **kw))
  File "/home/admin/.local/lib/python3.7/site-packages/prompt_toolkit/eventloop/coroutine.py", line 92, in step_next
    new_f = coroutine.throw(exc)
  File "/home/admin/.local/lib/python3.7/site-packages/prompt_toolkit/buffer.py", line 1506, in async_completer
    cancel=lambda: not proceed()))
  File "/home/admin/.local/lib/python3.7/site-packages/prompt_toolkit/eventloop/coroutine.py", line 88, in step_next
    new_f = coroutine.send(None)
  File "/home/admin/.local/lib/python3.7/site-packages/prompt_toolkit/eventloop/async_generator.py", line 117, in consume_async_generator
    item = iterator.send(send)
  File "/home/admin/.local/lib/python3.7/site-packages/prompt_toolkit/completion/base.py", line 176, in get_completions_async
    for item in self.get_completions(document, complete_event):
  File "/home/admin/.local/lib/python3.7/site-packages/athenacli/completer.py", line 210, in get_completions
    suggestions = suggest_type(document.text, document.text_before_cursor)
  File "/home/admin/.local/lib/python3.7/site-packages/athenacli/packages/completion_engine.py", line 121, in suggest_type
    full_text, identifier)
  File "/home/admin/.local/lib/python3.7/site-packages/athenacli/packages/completion_engine.py", line 293, in suggest_based_on_last_token
    tables = tuple(t for t in tables if identifies(parent, t))
  File "/home/admin/.local/lib/python3.7/site-packages/athenacli/packages/completion_engine.py", line 293, in <genexpr>
    tables = tuple(t for t in tables if identifies(parent, t))

Exception identifies() missing 2 required positional arguments: 'table' and 'alias'
```

I also tried this without the table aliases, e.g.:

```
SELECT doodads.* FROM doodads JOIN trinkets on doodads.
```

and got the same "unhandled exception" error when typing the `.` at the end.

### PR DESCRIPTION
**Pull request by pgr0ss**
Previously, the code passed the wrong arguments to `identifiers`.

Fixes #40.

This is my first time working in this project, so any and all feedback is welcome!

### CODE REVIEW

File: test/test_completion_engine.py [hunk 1]
```diff
@@ -28,6 +28,15 @@ def test_select_suggests_cols_with_qualified_table_scope():
...
 
+def test_join_suggests_cols_with_qualified_table_scope():
+    expression = 'SELECT * FROM tabl a JOIN tabl b on a.'
+    suggestions = suggest_type(expression, expression)
+
+    assert suggestions == (
+        Column(tables=((None, 'tabl', 'a'),), drop_unique=None),
+        Table(schema='a'),
+        View(schema='a'),
+        Function(schema='a', filter=None))
```
Comment (pgr0ss) [inline test/test_completion_engine.py]: I'm not super confident about this test. Did I do it correctly?

**Comment by getaaron** [comment]
I think we also want to add `JOIN` (and maybe `ON`?) to [`athenacli/packages/literals/literals.json`](https://github.com/dbcli/athenacli/blob/master/athenacli/packages/literals/literals.json)

**Comment by pgr0ss** [comment]
> I think we also want to add `JOIN` (and maybe `ON`?) to [`athenacli/packages/literals/literals.json`](https://github.com/dbcli/athenacli/blob/master/athenacli/packages/literals/literals.json)

I think that's for completing the word `JOIN` itself? Which we can certainly add, but doesn't seem related to the bug this PR fixes? Unless I'm misunderstanding.

**Comment by getaaron** [comment]
> > I think we also want to add `JOIN` (and maybe `ON`?) to [`athenacli/packages/literals/literals.json`](https://github.com/dbcli/athenacli/blob/master/athenacli/packages/literals/literals.json)
> 
> I think that's for completing the word `JOIN` itself? Which we can certainly add, but doesn't seem related to the bug this PR fixes? Unless I'm misunderstanding.

It's the first part:

> JOIN did not auto-complete as an option (I only saw JSON)

but I can move that out to a separate issue :+1:

**Comment by pgr0ss** [comment]
Ah, sorry, I missed that part.

File: test/test_completion_engine.py (hunk 1 above)
Comment (zzl0) [reply test/test_completion_engine.py]: Yes, it is correct

**Comment by getaaron** [comment]
@pgr0ss I added the keywords in https://github.com/dbcli/athenacli/pull/44.

**Comment by zzl0** [comment]
@pgr0ss We appreciate your help and want to give you credit. Please take a moment to put an `x` in the boxes below as you complete them. 

- [ ] I've added this contribution to the `changelog.md`.
- [ ] I've added my name to the `AUTHORS` file (or it's already there).

**Comment by pgr0ss** [comment]
@zzl0 I don't seem to be able to check those boxes, but I've done both.

**Comment by zzl0** [comment]
> @zzl0 I don't seem to be able to check those boxes, but I've done both.

It's fine, thanks

**Pull request merged**